### PR TITLE
Tracing

### PR DESCRIPTION
**Pull request by Bernardstanislas**
## What has been done

- setup tracing with opentelemetry and Datadog, which consists in sending traces to the Datadog APM
- remove slog for structured logging
- drain all logs to the traces, logs are then transformed in "events"
- events are output to the stdout, spans are sent to datadog

## What's left to do (in other PRs)

- append the span_id and trace_id in the logs for Datadog to link them together
- update the heroku buildpack configuration to include the Datadog agent and actually send the traces to Datadog
- update the log parsing in Datadog to adapt to the new format standardized by tracing_formatter

### CODE REVIEW

File: out.log [hunk 1]
```diff
@@ -0,0 +1,106 @@
+Oct 03 19:16:25.638 INFO This is my INFO trace!, span_id: 0000000000000000, trace_id: 00000000000000000000000000000000, commit: local_development
```
Comment (lechinoix) [inline out.log:1]: This file should be in .gitignore no ?

File: marketplace-infrastructure/src/logger/async.rs [hunk 2]
```diff
@@ -1,10 +0,0 @@
-use slog::{Drain, Never};
```
Comment (lechinoix) [inline marketplace-infrastructure/src/logger/async.rs:1]: Do we really want to delete all other loggers than tracing ?

File: marketplace-core/src/main.rs [hunk 3]
```diff
@@ -1,24 +1,13 @@
 use anyhow::Result;
 use dotenv::dotenv;
-use marketplace_infrastructure::logger;
-use slog::{Drain, Level, Logger};
+use marketplace_infrastructure::tracing::Tracer;
 
-fn create_root_logger() -> Logger {
```
Comment (lechinoix) [inline marketplace-core/src/main.rs:6]: We really want to suppress all other logs ?

File: out.log (hunk 1 above)
Comment (Bernardstanislas) [reply out.log]: whoops for sure

File: marketplace-infrastructure/src/logger/async.rs (hunk 2 above)
Comment (Bernardstanislas) [reply marketplace-infrastructure/src/logger/async.rs:1]: Actually we don't delete them, we just transform them into events with this https://github.com/onlydustxyz/marketplace-backend/pull/294/files#diff-ede4a145e0031e4ee6c8e12e1411ae337d046d3cf09df4aa1fbf370ce52bd9eaR34

File: marketplace-core/src/main.rs (hunk 3 above)
Comment (Bernardstanislas) [reply marketplace-core/src/main.rs:6]: See previous answer :)

File: marketplace-infrastructure/src/tracing/mod.rs [hunk 4]
```diff
@@ -9,38 +9,44 @@ use tracing_subscriber::{layer::SubscriberExt, EnvFilter};
...
+
+		LogTracer::init()?;
+
+		Ok(Tracer {})
+	}
 }
 
-pub fn teardown_tracing() {
-	shutdown_tracer_provider();
+impl Drop for Tracer {
```
Comment (AnthonyBuisset) [inline marketplace-infrastructure/src/tracing/mod.rs:40]: RAII ❤️

**Review by AnthonyBuisset — APPROVED** [review]

**Pull request merged**